pysmt/solvers/bdd.py

```diff
@@ -0,0 +1,324 @@
+        # operations on the model (e.g., enumeration) in a simple way.
+        if self.latest_model is None:
+            assert_ddmanager(self.ddmanager)
+            # Build ddArray of variables
+            var_array = self.converter.get_all_vars_array()
+            _, current_state = self.assertions_stack[-1]
+            minterm_set = current_state.PickOneMinterm(var_array, len(var_array))
+            minterm = next(iter(minterm_set))
+            assignment = {}
+            for i, node in enumerate(var_array):
+                value = self.mgr.Bool(minterm[i] == 1)
+                key = self.converter.node2var[node]
+                assignment[key] = value
+            self.latest_model = EagerModel(assignment=assignment,
+                                           environment=self.environment)
+        return self.latest_model
+
+    @clear_pending_pop
+    def push(self, levels=1):
+        for _ in xrange(levels):
+            self.backtrack.append(len(self.assertions_stack))
+
+    @clear_pending_pop
+    def pop(self, levels=1):
+        for _ in xrange(levels):
+            l = self.backtrack.pop()
+            self.assertions_stack = self.assertions_stack[:l]
+
+    def exit(self):
+        raise NotImplementedError
+
+    def __del__(self):
+        del self.ddmanager
+
+
+class BddConverter(DagWalker):
+
+    def __init__(self, environment, ddmanager):
+        DagWalker.__init__(self)
+
+        self.environment = environment
+        self.fmgr = self.environment.formula_manager
+        self.ddmanager = ddmanager
+        # Note: Nodes in pycudd are not shared, but they overload all
+        # methods to perform comparison. This means that for two
+        # wrappers for the variable x, we have that id(x) != id(x1)
+        # but x == x1.
+        self.node2var = {}
+        self.var2node = {}
+
+    def convert(self, formula):
+        """Convert a PySMT formula into a BDD."""
+        assert_ddmanager(self.ddmanager)
+        return self.walk(formula)
+
+    def back(self, bdd_expr):
+        assert_ddmanager(self.ddmanager)
+        return self.bdd_to_expr3(bdd_expr).simplify()
+
+    def get_all_vars_array(self):
+        # NOTE: This way of building the var_array does not look
+        #       robust.  There might be an issue if variables are
+        #       added and the order of enumeration of the dictionary
+        #       changes and we rely on this order outside of this class.
+        var_array = pycudd.DdArray(len(self.node2var))
+        for i, node in enumerate(self.node2var):
+            var_array[i] = node
+        return var_array
+
+    def cube_from_var_list(self, var_list):
+        assert_ddmanager(self.ddmanager)
+        indices = pycudd.IntArray(len(var_list))
+        for i, v in enumerate(var_list):
+            indices[i] = self.var2node[v].NodeReadIndex()
+        cube = self.ddmanager.IndicesToCube(indices, len(var_list))
+        return cube
+
+    @deprecated("convert")
```

Comment:
This is new, so we can avoid to create deprecated code, or is there a specific reason?